fastlane/fastlane
# [deliver] Implements `verify` with `altool` for Xcode 14 validation

### Checklist

### Motivation and Context
As part of our CI, we use `deliver`'s `--verify_only` flag (implemented by myself a few months back) to validate that our binary is fit to be uploaded to the App Store every night. 

### Description
Since the removal of the Transporter app from the Xcode 14.0 toolchain, this has stopped working for us. I have seen that some changes  have already been implemented by @freddi-kit and released to migrate some of `deliver` and `pilot`'s functionalities to using `altool`.

This PR builds on top of those changes and adds missing functionality to make the `--verify_only` flag work with `altool` by using its `--validate-app` option.

### Testing Steps
1. Archive an app and export an IPA with an invalid (for upload) version of Xcode. An example of this would be Xcode 14.1 beta 2.
2. Run deliver with the `verify_only` flag: `bundle exec fastlane deliver --ipa your_awesome_app.ipa --verify_only --verbose`
3. Validation should fail with the following error
<img width="1410" alt="CleanShot 2022-10-11 at 22 54 39@2x" src="https://user-images.githubusercontent.com/24246926/195205492-794c2ba0-6ea7-436b-a693-5ee7bc0f4eac.png">

@joshdholtz @freddi-kit Feel free to give me a shout if you need extra context or if there is something you'd like me to change 👍
